src/DistributedComputations/distributed_architectures.jl

```diff
@@ -51,10 +23,46 @@ Return `Partition` representing the division of a domain into
 `Rx` parts in `x` and `Ry` parts in `y` and `Rz` parts in `z`,
 where `x, y, z` are the first, second, and third dimension
 respectively.
+
+`Rx`, `Ry` and `Rz` can be vectors containing the percentage of the domain
+ascribed to the different cores. In this case, `length(Rx)` represent the number
+of divisions in the first dimension with `sum(Rx)` equal to 1.0. 
+Rank `arch.local_index[i]` will have `global_size[i] * Rx[i]` elements in the first dimension.
 """
-Partition(Rx::Number, Ry::Number=1, Rz::Number=1) = Partition{Rx, Ry, Rz}()
-Partition(lengths::Array{Int, 1}) = Partition(reshape(lengths, length(lengths), 1))
-Base.size(::Partition{<:Any, Rx, Ry, Rz}) where {Rx, Ry, Rz} = (Rx, Ry, Rz)
+Partition(Rx)     = Partition(validate_partition(Rx, 1, 1)...)
+Partition(Rx, Ry) = Partition(validate_partition(Rx, Ry, 1)...)
+
+function Partition(; x = 1, y = 1, z = 1)
+    x, y, z = validate_partition(x, y, z)
+    return Partition(x, y, z)
+end
+
+struct Equal end
+struct Fractional{S} end
+struct Sizes{S} end
```

Comment:
Why `Sizes{S}` and not

```julia
struct Sizes{S}
    sizes :: S
end
```

it's a little simpler if the struct can be interrogated via `partition.x.sizes`?

Reply:
ditto for `Fractional`